camel/models/nvidia_model.py

```diff
@@ -0,0 +1,188 @@
+        """
+        for param in self.model_config_dict:
+            if param not in NVIDIA_API_PARAMS:
+                raise ValueError(
+                    f"Unexpected parameter '{param}' for NVIDIA API."
+                )
+
+    def token_counter(self, text: str) -> int:
+        r"""Count the number of tokens in the given text.
+
+        Args:
+            text (str): The text to count tokens for.
+
+        Returns:
+            int: The number of tokens in the text.
+        """
+        return self._token_counter.count_tokens(text)
+
+    @api_keys_required("NVIDIA_API_KEY")
+    def run(
+        self,
+        messages: List[Dict[str, str]],
+        stream: bool = False,
+    ) -> Union[ChatCompletion, Stream[ChatCompletionChunk]]:
+        r"""Run the model with the given messages.
+
+        Args:
+            messages (List[Dict[str, str]]): List of messages for the conversation.
+            stream (bool, optional): Whether to stream the response.
+                (default: :obj:`False`)
+
+        Returns:
+            Union[ChatCompletion, Stream[ChatCompletionChunk]]: The model's response,
+            either as a complete message or a stream of chunks.
+        """
+        # Override stream parameter if specified in config
+        if 'stream' in self.model_config_dict:
+            stream = self.model_config_dict['stream']
+
+        # Prepare API parameters
+        api_params = {
+            "model": str(self.model_type),
+            "messages": messages,
+            "stream": stream,
+            "max_tokens": self.model_config_dict.get('max_tokens', 1024),
+            "temperature": self.model_config_dict.get('temperature', 0.5),
+            "top_p": self.model_config_dict.get('top_p', 1.0),
+            "frequency_penalty": self.model_config_dict.get('frequency_penalty', 0.0),
+            "presence_penalty": self.model_config_dict.get('presence_penalty', 0.0),
+            "seed": self.model_config_dict.get('seed', 0),
+        }
+
+        # Add optional parameters if specified
+        if 'stop' in self.model_config_dict:
+            api_params['stop'] = self.model_config_dict['stop']
+
+        # Call the API
+        response = self._client.chat.completions.create(**api_params)
+        return response
+
+    def _stream_response(self, response: Stream[ChatCompletion]) -> Generator[str, None, None]:
+        """Process streaming response.
+
+        Args:
+            response: The streaming response from the model.
+
+        Yields:
+            The content of each chunk in the stream.
+        """
+        for chunk in response:
+            if chunk.choices[0].delta.content is not None:
+                yield chunk.choices[0].delta.content
+
+    def chat(
+        self,
+        messages: List[Dict[str, str]],
+        stream: Optional[bool] = None,
+    ) -> Union[str, Generator[str, None, None]]:
+        """Chat with the model.
+
```

Comment:
docstring format need to be checked, you can refer to https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md#guideline-for-writing-docstrings